### PR TITLE
[codegen] Add multiple operands and function support when using fp16 compilation

### PR DESCRIPTION
**Pull request by zxy844288792**
Thanks for contributing to TVM!   Please refer to guideline https://docs.tvm.ai/contribute/ for useful information and tips. After the pull request is submitted, please request code reviews from [Reviewers](https://github.com/dmlc/tvm/blob/master/CONTRIBUTORS.md#reviewers).

As discussed in https://discuss.tvm.ai/t/error-cuda-compilation-error/3816 and https://discuss.tvm.ai/t/relay-automatic-fp16-downcasting/3952/3?u=xyzhou
The cuda fp16 computing uses “cuda_fp16.h”, It does not support operations with volatile keywords. Also, max function I refered to this PR, but it is not updating for a month, I add the min function support as well. 

I also edit the test_op_level1.py to enable fp16 test case. I will edit more test_op file but I would like to gather some feedback.

### CODE REVIEW

**Comment by vinx13** [comment]
I'm not sure what caused the error on CI.
Instead of adding these functions, directly overriding CUDA codegen rules for `half` might be preferred since we also want to deal with `half2` and avoid repetition

**Comment by zxy844288792** [comment]
> The error might be caused by incorrect arch setting. You can use
> https://github.com/dmlc/tvm/blob/cffb4fba03ea582417e2630bd163bca773756af6/python/tvm/contrib/nvcc.py#L218-L238
> 
> 
> to conditionally skip the test on ci.
> Instead of adding these functions, directly overriding CUDA codegen rules for `half` might be preferred since we also want to deal with `half2` and avoid repetition

Thanks for your information! I will try to use have_fp16 to skip the test on CI.  I will also start to investiagte how to override the codegen rules

**Comment by blacklong617** [comment]
I am similar to what you added, but there were some errors when I load resnet-18-fp16.onnx model on RTX2080. 
[cuda-got-error-cuda-error-launch-out-of-resources](https://discuss.tvm.ai/t/cuda-got-error-cuda-error-launch-out-of-resources/4173)

**Comment by tqchen** [comment]
@zxy844288792 @vinx13 please followup on this

**Comment by zxy844288792** [comment]
@vinx13 Can we get this PR merged with the current changes? We can take look at `half` and `half2` in a separate PR after we have more clarity.

File: tests/python/relay/test_op_level1.py [hunk 1]
```diff
@@ -130,226 +137,249 @@ def verify_expand_dims(dshape, dtype, oshape, axis, num_newaxis):
...
-        relay.ty.TensorType((3,), "float32")
-    ]))
+    for dtype in ['float16', 'float32']:
+        # beta and gamma ignored
+        data = relay.var("data", relay.TensorType((3, 2, 1), dtype))
+        beta = relay.var("beta", relay.TensorType((2,), dtype))
+        gamma = relay.var("gamma", relay.TensorType((2,), dtype))
+        moving_mean = relay.var("moving_mean", relay.TensorType((2,), dtype))
+        moving_var = relay.var("moving_var", relay.TensorType((2,), dtype))
+        y = relay.nn.batch_norm(data, gamma, beta, moving_mean, moving_var,
```
Comment (vinx13) [inline tests/python/relay/test_op_level1.py]: Fp16 for batch norm is not supported yet, need to merge #4088 first.

**Comment by vinx13** [comment]
@tqchen fp16 tests on ci are skipped now, any chances to get ci support for fp16 type?

**Comment by tqchen** [comment]
We will need to look into it, because most of the gpu workers we have do not yet have fp16 support. So we have to rely on manual check for now. I will see if we can get a fp16 enabled worker setup

**Comment by vinx13** [comment]
Thanks @zxy844288792 this is now merged

**Pull request merged**